### PR TITLE
test(analytics-api): unit tests for pure-logic modules (threshold/query/metric/config)

### PR DESCRIPTION
**Pull request by SharedQA**
Raises analytics-api unit coverage on the **pure-logic** modules (the part that doesn't need a DB/runtime).

## Measured (cargo-llvm-cov, instrumented)
- Workspace: **58.53% → 59.86% lines** (+1.33pts) from these tests. (api-gateway's larger jump is separate — #1327.)
- `threshold.rs` 0% → covered; `query.rs`/`metric.rs`/`config.rs` raised.
- 19 tests; analytics-api **257 pass**; `clippy --all-targets` clean (no unwrap/expect in tests).

## Test validity
Expected values are derived from **documented intent** — OData semantics, the in-code PATCH triple-state comment, the `default_*` fns — **not** from running the code. Negative/boundary cases included (unknown operator, missing required config, null-vs-absent PATCH, empty/oversized inputs).

**One unit was found semantically suspect during review and flagged rather than locked in:** `threshold_matches`'s `eq` uses `f64::EPSILON` (magnitude-dependent equality) — see **#1332**. The test characterises current behaviour; the issue asks for the intended rule.

## Coverage ceiling
Further unit gains are limited: the remaining uncovered analytics-api/oidc/clickhouse code is runtime/IO (`validate_token` needs JWKS, DB clients need ClickHouse/MariaDB, `main`/module wiring need the modkit runtime). Reaching ~80% requires the **instrumented e2e profile** (the rig running instrumented binaries — milestone 2.15), not more unit tests.

Related correctness findings from this pass: #1330 (duplicate/unstable data), #1331 (time formats), #1332 (threshold eq).

## Summary by CodeRabbit

* **Tests**
  * Added unit test coverage for analytics configuration precedence and required-field validation.
  * Added serialization/deserialization tests for metric update PATCH semantics and query/batch request behavior, including defaulting and JSON status formatting.
  * Added threshold evaluation and request deserialization tests covering supported operators, epsilon equality, invalid operator/level handling, and partial updates.

### CODE REVIEW

**Comment by SharedQA** [comment]
Overlaps #1314 (migrate to gears) on `domain/query.rs`. Reconcile these unit tests with the gears changes once #1314 merges.

File: src/backend/services/analytics-api/src/domain/metric.rs [hunk 1]
```diff
@@ -80,3 +80,69 @@ pub struct TableColumn {
...
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    type R = Result<(), Box<dyn std::error::Error>>;
+
+    #[test]
+    fn update_description_is_triple_state() -> R {
```
Comment (cyberantonz) [inline src/backend/services/analytics-api/src/domain/metric.rs:97]: Not sure what this test for. Put something to structure and check it is present? Strange idea.

File: src/backend/services/analytics-api/src/domain/metric.rs [hunk 2]
```diff
@@ -80,3 +80,69 @@ pub struct TableColumn {
...
+            id: Uuid::nil(),
+            insight_tenant_id: Uuid::nil(),
+            name: "m".to_owned(),
+            description: None,
+            query_ref: "SELECT 1".to_owned(),
+            is_enabled: true,
+            created_at: ts,
+            updated_at: ts,
+        };
+        let json = serde_json::to_string(&m)?;
```
Comment (cyberantonz) [inline src/backend/services/analytics-api/src/domain/metric.rs]: What this test for? You do not sure that serde do what it should do? 
I bet the serde developers have such kind of tests in their test cases.

File: src/backend/services/analytics-api/src/domain/metric.rs [hunk 3]
```diff
@@ -80,3 +80,69 @@ pub struct TableColumn {
...
+            !json.contains("description"),
+            "None description omitted: {json}"
+        );
+        assert!(json.contains("\"query_ref\":\"SELECT 1\""));
+        Ok(())
+    }
+
+    #[test]
+    fn metric_summary_never_exposes_query_ref() -> R {
+        let s = MetricSummary {
```
Comment (cyberantonz) [inline src/backend/services/analytics-api/src/domain/metric.rs]:
```rust
pub struct MetricSummary {
    pub id: Uuid,
    pub name: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub description: Option<String>,
}
```

There is no query ref in the structure. What is this test for?

File: src/backend/services/analytics-api/src/domain/query.rs [hunk 4]
```diff
@@ -85,3 +85,73 @@ pub enum BatchQueryResult {
...
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    type R = Result<(), Box<dyn std::error::Error>>;
+
+    #[test]
+    fn default_top_is_25() {
```
Comment (cyberantonz) [inline src/backend/services/analytics-api/src/domain/query.rs]: Why this test exists? The function is just return 25 and we need to check it. It is strange

File: src/backend/services/analytics-api/src/config.rs [hunk 5]
```diff
@@ -98,3 +98,57 @@ impl AppConfig {
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    type R = Result<(), Box<dyn std::error::Error>>;
+
+    #[test]
+    fn default_helpers() {
+        assert_eq!(default_bind_addr(), "0.0.0.0:8081");
```
Comment (cyberantonz) [inline src/backend/services/analytics-api/src/config.rs]: Why we need to test those values at all?

**Review by cyberantonz — COMMENTED** [review]
Needs work

File: src/backend/services/analytics-api/src/domain/metric.rs (hunk 1 above)
Comment (SharedQA) [reply src/backend/services/analytics-api/src/domain/metric.rs:97]: That one was just round-tripping through serde — it couldn't fail unless serde itself broke. Removed in 27045b4f.

File: src/backend/services/analytics-api/src/domain/metric.rs (hunk 2 above)
Comment (SharedQA) [reply src/backend/services/analytics-api/src/domain/metric.rs]: No point re-testing serde. Dropped it. The only metric test I kept is the triple-state one: description is Option<Option<_>> with a custom deserializer (absent / null / value), so that branching is actually ours, not the library's.

File: src/backend/services/analytics-api/src/domain/metric.rs (hunk 3 above)
Comment (SharedQA) [reply src/backend/services/analytics-api/src/domain/metric.rs]: Good point — MetricSummary doesn't even have a query_ref field, so the assert was dead weight. Removed.

File: src/backend/services/analytics-api/src/domain/query.rs (hunk 4 above)
Comment (SharedQA) [reply src/backend/services/analytics-api/src/domain/query.rs]: Agreed, asserting default_top() returns 25 proves nothing. Gone. The default still gets exercised through real deserialization in query_request_applies_defaults_when_empty.

File: src/backend/services/analytics-api/src/config.rs (hunk 5 above)
Comment (SharedQA) [reply src/backend/services/analytics-api/src/config.rs]: Same here — that was just testing the defaults equal their literals. Removed default_helpers and the defaults-extraction test; kept only the override-precedence and missing-clickhouse_url cases, which check real behavior.

**Comment by ktursunov** [comment]
Already covered with unit tests coverage
